keps/1145-additional-labels/README.md

```diff
@@ -0,0 +1,152 @@
+admitting may either narrow down on which machines the workload should start (by adding
+some extra node selector/tolerations) or add some tracking information
+that may be needed by other system components outside of Kueue.
+
+### Goals
+
+* Establish a mechanism through wich `AdmissionChecks` can add annotations/labels/node selector/tolerations
+  to workload pods.
+
+### Non-Goals
+
+* Allow `AdmissionChecks` to change pod definition in other way, for instance change command or image name.
+* Hard code any specific labels/annotations that are passed down.
+
+## Proposal
+
+* Add a struct in `Workload` status that allows to set annotations, labels, taints and tolerations on PodSets.
+* Expend PodSetInfo in Integration Framework to pass this information.
+* Modify existing integrations to support setting more data on the pods.
+
+### User Stories (Optional)
+
+#### Story 1
+
+My `AdmissionCheck` preallocates specific nodes (tainted and labeled) for a Workload to run, before it is 
+admitted (and creates any worker pods). Workload pods need to be modified so that they end up on correct nodes.
+
+#### Story 2
+
+My `AdmissionCheck` does budget checking and would like to mark pods that are possibly crossing budget with a 
+specific label for easier tracking and alerting.
+
+### Risks and Mitigations
+
+Some integrations may not support setting all of the desired information due to their CRD limitations.
+
+## Design Details
+
+Modify WorkloadStatus API object:
+
+```
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+    [...]
+    AdmissionChecks []metav1.Condition `json:"admissionChecks,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+    PodSetUpdates []PodSetUpdate `json:"podSetUpdates,omitempty"
+}
+
+// PodSetUpdate contains a list of pod set modifications suggested by AdmissionChecks.
+// The modifications should be additive only - modifications of already existing keys
+// are not allowed and will result in failure during workload admission.
+type PodSetUpdate struct {
+    // Name of the PodSet to modify. Should match to one of the Workload's PodSets.
+    Name string `json:"name"`
+
+    // +optional
+    Labels map[string]string `json:"labels,omitempty"
+
+    // +optional
+    Annotations map[string]string `json:"annotations,omitempty" 
+
+    // +optional
+    NodeSelector map[string]string `json:"nodeSelector,omitempty"
+
+    // +optional
+    Tolerations []Toleration `json:"tolerations,omitempty"
+}
+```
+
+Each of the `AdmissionChecks` controllers should populate the needed fields before 
+(or at the same time) flipping the corresponding `AdmissionCheck` to `True`. The values
+from `PodSetUpdates` are combined with `ResourceFlavors` into integration framework
+`PodSetInfo` (that will get the missing labels, tolerations and annotations fields) 
+and then passed to the framework controller for application on the CRD.
```

Comment:
I would like to see two things discussed:
1. how the information is restored, say the batch.Job, when we suspend the workload, so that potentially new labels, annotations, and tollerations can be assigned on re-admission
2. how about single pods support? I suppose we cannot mutate them, and there is no CRD to manage them, so IIUC we will need to recreate them, thus supporting them might require extra effort.